File: src/main/java/hu/frontrider/arcana/registrationhandlers/CreatureEnchantRegistry.java
package hu.frontrider.arcana.registrationhandlers;

import hu.frontrider.arcana.creatureenchant.backend.CreatureEnchant;
import hu.frontrider.arcana.creatureenchant.backend.EnchantingBaseCircle;
import hu.frontrider.arcana.creatureenchant.base.NegationCircle;
import hu.frontrider.arcana.creatureenchant.base.NormalCircle;
import hu.frontrider.arcana.creatureenchant.effect.*;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.RegistryEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.registries.RegistryBuilder;

import static hu.frontrider.arcana.ThaumicArcana.MODID;


@Mod.EventBusSubscriber
public class CreatureEnchantRegistry {

    @SubscribeEvent
    public static void register(RegistryEvent.NewRegistry newRegistry) {
        new RegistryBuilder<CreatureEnchant>()
                .setType(CreatureEnchant.class)
                .setIDRange(0, 100)
                .setName(new ResourceLocation(MODID,"creature_enchant"))
                .add((iForgeRegistryInternal, registryManager, i, enchant, v1) -> MinecraftForge.EVENT_BUS.register(enchant))
                .create();

        new RegistryBuilder<EnchantingBaseCircle>()
                .setType(EnchantingBaseCircle.class)
                .setIDRange(0, 100)
                .setName(new ResourceLocation(MODID,"enchanting_base_circle"))
                .create();
    }

    @SubscribeEvent
    public static void registryEventCreatureEnchant(RegistryEvent.Register<CreatureEnchant> event) {
        event.getRegistry().registerAll(
                new StrengthEnchant(),
                new FertileEnchant(),
                new ProtectionEnchant(),
                new RespirationEnchant(),
                new SpiderFingers(),
                new Speed(),
                new Vitality()
        );
    }
    @SubscribeEvent
    public static void registryEventEnchantingBaseCircle(RegistryEvent.Register<EnchantingBaseCircle> event) {
        event.getRegistry().registerAll(
                new NormalCircle(),
                new NegationCircle()
        );
    }

}

File: src/main/java/hu/frontrider/arcana/registrationhandlers/CreatureEnchantRegistry.kt
package hu.frontrider.arcana.registrationhandlers

import hu.frontrider.arcana.creatureenchant.base.EffectAmplifierCircle
import hu.frontrider.arcana.creatureenchant.base.NegationCircle
import hu.frontrider.arcana.creatureenchant.base.NormalCircle
import hu.frontrider.arcana.creatureenchant.effect.*
import hu.frontrider.arcana.creatureenchant.CreatureEnchant
import hu.frontrider.arcana.creatureenchant.EnchantingBaseCircle
import net.minecraftforge.event.RegistryEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class CreatureEnchantRegistry {

    companion object {

        var STRENGTH_ENCHANT: CreatureEnchant = StrengthEnchant()
        var FERTILE_ENCHANT: CreatureEnchant = FertileEnchant()
        var PROTECTION_ENCHANT: CreatureEnchant = ProtectionEnchant()
        var RESPIRATION_ENCHANT: CreatureEnchant = RespirationEnchant()
        var SPIDERFINGERS_ENCHANT: CreatureEnchant = SpiderFingers()
        var SPEED_ENCHANT: CreatureEnchant = Speed()
        var VITALITY_ENCHANT: CreatureEnchant = Vitality()

    }

    @SubscribeEvent
    fun registryEventCreatureEnchant(event: RegistryEvent.Register<CreatureEnchant>) {
        event.registry.registerAll(
                STRENGTH_ENCHANT, SPEED_ENCHANT, SPIDERFINGERS_ENCHANT,
                PROTECTION_ENCHANT, FERTILE_ENCHANT, RESPIRATION_ENCHANT, VITALITY_ENCHANT,
                HealtBoostEnchant(),
                AgilityEnchant(),
                HasteEnchant(),
                ProductivityEnchant()
        )
    }

    @SubscribeEvent
    fun registryEventEnchantingBaseCircle(event: RegistryEvent.Register<EnchantingBaseCircle>) {
        event.registry.registerAll(
                NormalCircle(),
                NegationCircle(),
                EffectAmplifierCircle()
        )
    }

}

